app/recipe/serializers.py

```diff
@@ -0,0 +1,54 @@
+from rest_framework import serializers
+
+from core.models import Recipe, Ingredient
+
+
+class IngredientSerializer(serializers.ModelSerializer):
+    """Serializer for the Ingredient object"""
+
+    class Meta:
+        model = Ingredient
+        fields = ('id', 'name', )
+        read_only_fields = ('id', )
+
+
+class RecipeSerializer(serializers.ModelSerializer):
+    """Serializer for the Recipe object"""
+    ingredients = IngredientSerializer(
+        many=True,
+        required=False
+    )
+
+    class Meta:
+        model = Recipe
+        fields = ['id', 'name', 'description', 'ingredients']
+        read_only_fields = ('id', )
+
+    def create(self, validated_data):
+        """Create recipe related with the ingredients"""
+        ingredients = validated_data.pop('ingredients', [])
+        recipe = Recipe.objects.create(**validated_data)
+
+        for ingredient in ingredients:
+            Ingredient.objects.create(
+                name=ingredient['name'],
+                recipe=recipe
+            )
+
+        return recipe
+
+    def update(self, instance, validated_data):
+        """Update recipe related with the ingredients"""
+        ingredients = validated_data.pop('ingredients', None)
```

Comment:
```suggestion
        ingredients = validated_data.pop('ingredients', [])
```
Like this you could avoid the check in line 45 for `ingredients` being `None`.

Reply:
Ok